### PR TITLE
Force username & email to lowerCase PLUS Remove sensitive data before…

### PR DESCRIPTION
**Pull request by almegdad**
… return authenticated user in password reset

### CODE REVIEW

**Comment by lirantal** [comment]
@rhutchison @codydaig @ilanbiala @mleanos any thoughts?

File: modules/users/client/views/authentication/signin.client.view.html [hunk 1]
```diff
@@ -6,7 +6,7 @@ <h3 class="col-md-12 text-center">Or with your account</h3>
         <div class="form-group">
           <label for="username">Username</label>
           <input type="text" id="username" name="username" class="form-control"
-               data-ng-model="credentials.username" placeholder="Username">
+               data-ng-model="credentials.username" data-ng-change="credentials.username=credentials.username.toLowerCase();" placeholder="Username">
```
Comment (lirantal) [inline modules/users/client/views/authentication/signin.client.view.html]: We seem to have decided not to use the `data-` prefix for angular attributes so please change this through-out the PR to just `ng-change` and so on..

**Comment by rhutchison** [comment]
can we use css `text-transform: lowercase;` instead of ng-change, and then on save (client or server) force to lower?

**Comment by mleanos** [comment]
I agree with @rhutchison that this should be done with the save method, either on the client or server side. I'm not a fan of using `ng-change` for these types of scenarios. 

I don't see a need to transform with css. The real need here is to change the username to lowercase, *sometime*, before authentication; in order to decrease the chance of user error. 

However, if the view must be updated to reflect the lowercase change, then I'd say that's an argument to do the conversion, in the client controller `save` method; thus, updating the view.

**Comment by mleanos** [comment]
After looking at these changes a bit further, I think the lowercase conversion should be done at the very front-end with the user input; most likely in the client controller method that initiates any API calls. If these changes are made in the middleware, they can get lost & the SOC gets weaker.

To ensure the username & email always get converted to lowercase, the pre-`save` hook can be used on the User model.

**Comment by rhutchison** [comment]
>I don't see a need to transform with css. The real need here is to change the username to lowercase, sometime, before authentication; in order to decrease the chance of user error.

>However, if the view must be updated to reflect the lowercase change, then I'd say that's an argument to do the conversion, in the client controller save method; thus, updating the view.

Why wouldn't you just have text-transform?  the user doesn't need to know when it's transformed, just that it is being transformed.

**Comment by mleanos** [comment]
@rhutchison I don't like using css as a means to transform data based on business logic/needs. The client/server controller would be sufficient in most cases. If someone wants to update the view to show the transformation, then adding the css `text-transform` would be an enhancement, IMO.

**Comment by ilanbiala** [comment]
.toLowerCase is good, but I'd do it in the JS and not in the HTML.

**Comment by simison** [comment]
Those ng-change bits should be removed; after that this LGTM and pretty much how I'm doing it already at my copy.

+1 @mleanos - CSS/angular frontend tweaking seems hacky.

**Comment by almegdad** [comment]
reopened #866

**Pull request closed without merging**